server/src/main/java/org/apache/druid/curator/CuratorModule.java

```diff
@@ -79,10 +82,33 @@ public CuratorFramework makeCurator(CuratorConfig config, EnsembleProvider ensem
           StringUtils.format("%s:%s", config.getZkUser(), config.getZkPwd()).getBytes(StandardCharsets.UTF_8)
       );
     }
+
+    RetryPolicy retryPolicy;
+    if (config.getTerminateDruidProcessOnConnectFail()) {
+      final Runnable exitRunner = () -> {
+        try {
+          log.error("Zookeeper can't be reached, forcefully stopping lifecycle...");
+          lifecycle.stop();
+          System.err.println("Zookeeper can't be reached, forcefully stopping virtual machine...");
+        }
+        finally {
+          System.exit(1);
```

Comment:
I believe you cannot write a unit-test for that.

Reply:
You're basically right.

Without adding a bunch of PowerMockito dependencies for catching a static `System.exit(..)` call (which I don't want to do), I simulated it in a unit test by throwing an unchecked RuntimeException (instead of `System.exit(...)`) from an identical block of code and catching that.